src/Command/Dedupe.php

```diff
@@ -74,6 +62,49 @@ protected function execute(InputInterface $input, OutputInterface $output): void
         }
     }
 
+    /**
+     * @param string[] $fields
+     */
+    private function getValuesHash(array $fields, array $data): string
+    {
+        $values = [];
+        foreach ($fields as $field) {
+            $values[$field] = $data[$field];
+        }
+
+        return sha1(serialize($values));
+    }
+
+    /**
+     * @param string[] $fields
+     * @param bool[]   $seenValues Hashmap of value-hash => bool, to keep track of values already encountered
+     */
```

Comment:
we should document the return value here i think: `@return array the $message if this is no duplicate, or null if $fields is in $seenValues`